### PR TITLE
update build model to use repo name

### PR DESCRIPTION
**Pull request by epananth**
https://github.com/dotnet/arcade/issues/6100

### CODE REVIEW

**Comment by riarenas** [comment]
Looks reasonable. How did you validate this?

**Comment by mmitche** [comment]
LGTM. FYI This will require a corresponding change in in windowsdesktop, runtime, aspnetcore and installer on the move to v3. But it should not chagne v2.

**Review by mmitche — APPROVED** [review]

**Comment by riarenas** [comment]
>Right now we don't have any test to create the manifest not sure if that is going to add value in the bigger picture. We have test after the manifest is created and all of that were successful.

You can do an internal build to see if the fields now look right in the manifest.

**Comment by epananth** [comment]
> LGTM. FYI This will require a corresponding change in in windowsdesktop, runtime, aspnetcore and installer on the move to v3. But it should not chagne v2.

Yes I will create a pr for Windows desktop right away, also I will add this as one of the steps to do while moving to V3 for other repos

**Comment by mmitche** [comment]
> > LGTM. FYI This will require a corresponding change in in windowsdesktop, runtime, aspnetcore and installer on the move to v3. But it should not chagne v2.
> 
> Yes I will create a pr for Windows desktop right away, also I will add this as one of the steps to do while moving to V3 for other repos

Please create the PR in WD against release/5.0-rc2

**Comment by epananth** [comment]
ManifestRepoName is not present in the task, so reverted the changes and just added a new param till we get arcade update with that param 

validated the changes -> 

image

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs [hunk 1]
```diff
@@ -43,6 +43,8 @@ public class PushToBlobFeed : MSBuild.Task
 
         public int UploadTimeoutInMinutes { get; set; } = 5;
 
+        public string ManifestRepoName { get; set; }
```
Comment (mmitche) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs:46]: Is this change required? Is anyone in 5.x using PushToBlobFeed any longer? Btw, we should remove this task altogether as part of the cleanup epic.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs [hunk 2]
```diff
@@ -20,6 +20,8 @@ public class PushToAzureDevOpsArtifacts : MSBuild.Task
 
         public bool PublishFlatContainer { get; set; }
 
+        public string ManifestRepoName { get; set; }
```
Comment (mmitche) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs:23]: Does it make sense to change this task to use ManifestRepoName if it's non-empty. This will give us a way to move repos to using ManifestRepoName instead of ManifestRepoUri, then eventually removing ManifestRepoUri.

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs [hunk 3]
```diff
@@ -173,7 +173,7 @@ public override bool Execute()
                         blobArtifacts,
                         packageArtifacts,
                         AssetManifestPath,
-                        ManifestRepoUri,
+                        ManifestRepoName != null? ManifestRepoName : ManifestRepoUri,
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToAzureDevOpsArtifacts.cs:176]: Maybe check for String.IsNullOrEmpty here?

**Review by riarenas — APPROVED** [review]

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs (hunk 1 above)
Comment (epananth) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs]: I don't any of the 5.x repos are using PushToBlob, but I will check again as part of this issue -> https://github.com/dotnet/arcade/issues/6117

**Comment by mmitche** [comment]
A quick GH search indicates it looks like only a couple 3.x repos (e.g. msbuild)

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs [hunk 4]
```diff
@@ -1,5 +1,6 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
```
Comment (chcosta) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/PushToBlobFeed.cs:3]: @epananth , I presume this was unintentionally added back?
